src/internal/EnhancedButton.js

```diff
@@ -318,12 +327,12 @@ class EnhancedButton extends Component {
     };
     const buttonChildren = this.createButtonChildren();
 
-    // Provides backward compatibity. Added to support wrapping around <a> element.
-    const targetLinkElement = buttonProps.hasOwnProperty('href') ? 'a' : 'span';
+    // Provides backwards compatibility. Added to support wrapping around <a> element.
+    const targetLinkElement = href ? 'a' : 'span';
 
     return React.isValidElement(containerElement) ?
       React.cloneElement(containerElement, buttonProps, buttonChildren) :
-      React.createElement(linkButton ? targetLinkElement : containerElement, buttonProps, buttonChildren);
+      React.createElement((href || listItem) ? targetLinkElement : containerElement, buttonProps, buttonChildren);
```

Comment:
The purpose of of the `linkButton` prop was to signify that there was an href, and so set the container as `<a>`.

`ListItem` was abusing that prop to set it span, but tracing the code, I think we should just be able to use the `containerElement` prop for that in `ListItem`.

I'll give it a go.